src/plugins/data/common/search/search_source/search_source.test.ts

```diff
@@ -1117,8 +1117,10 @@ describe('SearchSource', () => {
 
         searchSource = new SearchSource({}, searchSourceDependencies);
         searchSource.setField('index', indexPattern);
+
         await searchSource
           .fetch$(options)
+          .pipe(catchError((e) => e))
```

Comment:
I played around with this code for a bit hoping to find a solution that was a bit more direct. Why do you need this line when `catch` is called below? I approve of the PR but wish I understood this a bit better.

Reply:
I think the problem in this test is actually a few lines up above:

```ts
searchSourceDependencies.search = jest.fn().mockReturnValue(of(Promise.reject('aaaaa')));
```

The mock `search` function will throw an uncaught error as soon as `fetch$` is called (rather than when it's subscribed to, as the real `search` function would). In fact, if you keep this line as-is and get rid of the [rest of this test](https://github.com/elastic/kibana/blob/de43a3b83d90d66744d9f71fbe4c742bed9df6af/src/plugins/data/common/search/search_source/search_source.test.ts#L1118-L1129), then run the test with the flag, you'll still get the unhandled rejection.

Instead, I think we want to change this line to the following:

```ts
searchSourceDependencies.search = jest.fn().mockReturnValue(throwError('aaaaa'));
```

If we make this change, the rest of the test can remain unchanged and we won't see the given error.

Reply:
Thanks for taking a closer look @mattkime , I was also a bit confused! And thanks for the explanation @lukasolson ! I'll update the PR.

Reply:
nice work @lukasolson !